### PR TITLE
Feature/json api secondary tags switch

### PR DESCRIPTION
**Pull request by jonhindle**
### Context

> Does this issue have a Trello card?

https://trello.com/c/y9lvSy1U/111-switch-to-jsonapi-for-secondary-tags

> If this is an issue, do we have steps to reproduce?

N/A

### Intent

> What changes are introduced by this PR that correspond to the above card?

Moving to using the JSONApi

> Would this PR benefit from screenshots?

N/A

### Considerations

> Is there any additional information that would help when reviewing this PR?

N/A

> Are there any steps required when merging/deploying this PR?

No

### Checklist

- [ ] This PR contains **only** changes related to the above card
- [ ] Tests have been added/updated to cover the change
- [ ] Documentation has been updated where appropriate
- [ ] Tested in Development

### CODE REVIEW

File: server/repositories/__tests__/cmsApi.spec.js [hunk 1]
```diff
@@ -145,7 +145,7 @@ describe('CmsApi', () => {
           }),
         ]),
       );
-      const response = await cmsApi.get(new TestQuery());
+      const response = await cmsApi.get(new TestUrltransformEachQuery());
```
Comment (andrewrlee) [inline server/repositories/__tests__/cmsApi.spec.js:148]: meh.. not camel case

File: server/repositories/cmsQueries/__tests__/secondaryTagHeaderPageQuery.spec.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    it('should create the correct url', async () => {
+      expect(query.url()).toStrictEqual(
+        'https://cms/content/1234?include=field_featured_image&fields%5Btaxonomy_term--tags%5D=name%2Cdescription%2Cdrupal_internal__tid%2Cfield_featured_image',
+      );
+    });
+  });
+
+  describe('transform', () => {
+    it('should create correct basic page structure', async () => {
+      const basicPage = {
```
Comment (andrewrlee) [inline server/repositories/cmsQueries/__tests__/secondaryTagHeaderPageQuery.spec.js:17]: meh.. presumably this isn't a basic page?

File: server/repositories/cmsQueries/__tests__/secondaryTagHeaderPageQuery.spec.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+const {
+  SecondaryTagHeaderPageQuery,
+} = require('../secondaryTagHeaderPageQuery');
+
+describe('Basic page query', () => {
```
Comment (andrewrlee) [inline server/repositories/cmsQueries/__tests__/secondaryTagHeaderPageQuery.spec.js:5]: wrong test name

File: server/repositories/cmsQueries/__tests__/secondaryTagPageQuery.spec.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      const SECONDARYTAG1 = createSecondaryTag(1);
+      const SECONDARYTAG2 = createSecondaryTag(UUID);
+      const SECONDARYTAGS = [SECONDARYTAG1, SECONDARYTAG2];
+      const CONTENT1 = createContent('A', SECONDARYTAGS);
+      const CONTENT2 = createContent('B', [SECONDARYTAG2]);
+      const CONTENT3 = createContent('C', SECONDARYTAGS);
+      const response = [CONTENT1, CONTENT2, CONTENT3];
+
+      expect(query.transform(response)).toStrictEqual(
+        Object.assign(createTransformedSecondaryTag(UUID), {
```
Comment (andrewrlee) [inline server/repositories/cmsQueries/__tests__/secondaryTagPageQuery.spec.js:66]: just a style question more than anything, why do you prefer this to splatting?

File: server/views/pages/tagsSeries.html [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    data: data.relatedContent,
+    lazyLoadContent: true
+  }) }}
+
+  <div class="govuk-form-group govuk-!-margin-bottom-2">
+    {% if data.contentType =='series'%}
+      <p class="govuk-hub-article-feedback govuk-hub-feedback-text">Tell us what you think about this series:</p>
+    {% endif %}
+
+    {% if data.contentType =='tags'%}
```
Comment (andrewrlee) [inline server/views/pages/tagsSeries.html:82]: guess this isn't needed any more, but NBD. We'll hopefully revisit this as part of the next card

File: server/repositories/cmsQueries/__tests__/secondaryTagPageQuery.spec.js (hunk 4 above)
Comment (jonhindle) [reply server/repositories/cmsQueries/__tests__/secondaryTagPageQuery.spec.js:66]: I tend to use it over splatting when all the new objects are transient. It was (maybe still is) marginally quicker and preserves the nesting from the original object.

File: server/repositories/cmsQueries/__tests__/secondaryTagHeaderPageQuery.spec.js (hunk 3 above)
Comment (jonhindle) [reply server/repositories/cmsQueries/__tests__/secondaryTagHeaderPageQuery.spec.js:5]: 🤠

File: server/repositories/cmsQueries/__tests__/secondaryTagHeaderPageQuery.spec.js (hunk 2 above)
Comment (jonhindle) [reply server/repositories/cmsQueries/__tests__/secondaryTagHeaderPageQuery.spec.js:17]: 🤠

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**